SwiftGit2/Repository.swift

```diff
@@ -537,4 +537,297 @@ final public class Repository {
+			                                oldFile: oldFile,
+			                                newFile: newFile)
+
+			returnDict.append(gitDiffDelta)
+
+			git_diff_free(OpaquePointer(delta))
+		}
+
+		let result = Result<[GitDiffDelta], NSError>.success(returnDict)
+		return result
+	}
+
+	// MARK: - Status
+
+	public func getRepositoryStatus() -> [GitStatusEntry] {
+
+		var returnArray = [GitStatusEntry]()
+
+		// Do this because GIT_STATUS_OPTIONS_INIT is unavailable in swift
+		let pointer = UnsafeMutablePointer<git_status_options>.allocate(capacity: 1)
+		git_status_init_options(pointer, UInt32(GIT_STATUS_OPTIONS_VERSION))
+		var options = pointer.move()
+		pointer.deallocate(capacity: 1)
+
+		var status: OpaquePointer? = nil
+		git_status_list_new(&status, self.pointer, &options)
+
+		let count = git_status_list_entrycount(status)
+
+		for i in 0..<count {
+			let s = git_status_byindex(status, i)
+			if s?.pointee.status.rawValue == GIT_STATUS_CURRENT.rawValue {
+				continue
+			}
+			var status: GitStatus? = nil
+
+			var headToIndex: GitDiffDelta? = nil
+			var htoiStatus: GitStatus? = nil
+			var htoiOldFile: GitDiffFile? = nil
+			var htoiNewFile: GitDiffFile? = nil
+
+			var indexToWorkDir: GitDiffDelta? = nil
+			var itowStatus: GitStatus? = nil
+			var itowOldFile: GitDiffFile? = nil
+			var itowNewFile: GitDiffFile? = nil
+
+			// Delta status
+			if let statusValue = s?.pointee.status.rawValue {
+				status = self.convertStatus(statusValue)
+			}
+
+			// Head To Index status and files
+			if s?.pointee.head_to_index != nil {
+				if let statusValue = s?.pointee.head_to_index.pointee.status.rawValue {
+					htoiStatus = self.convertStatus(statusValue)
+				}
+				if let oldFile = s?.pointee.head_to_index.pointee.old_file {
+					htoiOldFile = self.convertDiffFile(oldFile)
+				}
+				if let newFile = s?.pointee.head_to_index.pointee.new_file {
+					htoiNewFile = self.convertDiffFile(newFile)
+				}
+
+				headToIndex = GitDiffDelta(status: htoiStatus,
+				                           flags: s?.pointee.head_to_index.pointee.flags,
+				                           oldFile: htoiOldFile,
+				                           newFile: htoiNewFile)
+			}
+
+			// Index to Working Directory status and files
+			if s?.pointee.index_to_workdir != nil {
+				if let statusValue = s?.pointee.index_to_workdir.pointee.status.rawValue {
+					itowStatus = self.convertStatus(statusValue)
+				}
+				if let oldFile = s?.pointee.index_to_workdir.pointee.old_file {
+					itowOldFile = self.convertDiffFile(oldFile)
+				}
+				if let newFile = s?.pointee.index_to_workdir.pointee.new_file {
+					itowNewFile = self.convertDiffFile(newFile)
+				}
+
+				indexToWorkDir = GitDiffDelta(status: itowStatus,
+				                              flags: s?.pointee.index_to_workdir.pointee.flags,
+				                              oldFile: itowOldFile,
+				                              newFile: itowNewFile)
+			}
```

Comment:
The code here is duplicated with the code in the previous `if`. Can you add a function that creates a `GitDiffDelta` from a `git_diff_delta`? Probably adding an `init` to `GitDiffDelta` is the best way to do this.